### DISCUSSION TITLE
clip voronoi mesh to geo polygon

### DISCUSSION THREAD

**Discussion by pdebruic** [observablehq/plot#1338] [Q&A] [answered]
I notice in the docs the voronoi mesh is clipped to the frame.  Is there a way to clip it or mask it to a e.g. a US state border or county border? Thanks!

**Comment by mbostock** [comment]
You can do it in SVG, but there’s not a built-in way to do this with Plot currently. See https://github.com/observablehq/plot/issues/1109.

**Comment by pdebruic** [comment, marked as answer]
Thanks for the pointer.  Because I don't know how to incorporate the workflow from #1109 I ended up writing some js to post-process the plots. 

```
var geos = document.querySelectorAll('g[aria-label="geo"]')
document.querySelectorAll('g[aria-label="voronoi"]').forEach((element,index) =>{
    var clipPath = document.createElementNS('http://www.w3.org/2000/svg', 'clipPath');
    clipPath.innerHTML=geos[index].innerHTML;
    var clipId = 'clip'+index;
    clipPath.setAttribute('id',clipId);
    var plot = element.parentNode;
    plot.appendChild(clipPath);
    element.setAttribute('clip-path',"url(#"+clipId+")");
})
```

Which works for my needs.  

Is there a way to set a class or id attribute on a plot? 

<img width="530" alt="Screen Shot 2023-03-15 at 6 46 02 PM" src="https://user-images.githubusercontent.com/67050/225488777-34040091-8b3b-4870-a3b9-2cbd02929b7e.png">

**Reply by Fil** [reply]
There's no way currently to add a class; see https://github.com/observablehq/plot/issues/1002

**Reply by mbostock** [reply]
You can however set a class on the SVG element using the top-level **className** option.

**Reply by pdebruic** [reply]
Thanks!  Was strange to discover that a  camelCase class name gives an illegal class name error on Firefox.  e.g. `className: "myClass"` is no good.  I didn't test other browsers.

**Comment by pdebruic** [comment]
saner version of the above js
```

document.querySelectorAll('g[aria-label="voronoi"]').forEach((element,index) =>{
    var geo = element.parentNode.querySelector('g[aria-label="geo"]')
    if(geo){
    var clipPath = document.createElementNS('http://www.w3.org/2000/svg', 'clipPath');
    clipPath.innerHTML=geo.innerHTML;
    var clipId = 'clip'+index;
    clipPath.setAttribute('id',clipId);
    var plot = element.parentNode;
    plot.appendChild(clipPath);
    element.setAttribute('clip-path',"url(#"+clipId+")");
    }})
```
